src/main/java/fr/openent/competences/service/impl/DefaultNoteService.java

```diff
@@ -3649,21 +3671,37 @@ public void getDataGraphDomaine(final String idEleve, JsonArray groupIds, final
         competenceNoteService.getConversionNoteCompetence(idEtablissement, idClasse,tableauEvent ->
                 formate(tableauDeConversionFuture, tableauEvent));
 
-        // 4. On Lie les compétences-Notes à leur libellé
-        CompositeFuture.all(compNotesFuture, domainesCycleFuture,tableauDeConversionFuture).setHandler(event -> {
-            if (event.succeeded()) {
-                handler.handle(new Either.Right<>(linkCompNoteToLibelle(domainesCycleFuture.result(),
-                        compNotesFuture.result(), tableauDeConversionFuture.result(), idEleve)));
-            } else {
-                String message = event.cause().getMessage();
-                handler.handle(new Either.Left<>(message));
-                log.error(message);
-            }
+        Promise<Boolean> isAvgSkillpromise = Promise.promise() ;
+        structureOptionsService.getIsAverageSkills(idEtablissement, event -> {
+            if(event.isRight())
+                isAvgSkillpromise.complete(event.right().getValue().getBoolean("is_average_skills"));
+            else
+                isAvgSkillpromise.fail(event.left().getValue());
         });
 
+        isAvgSkillpromise.future()
```

Comment:
Voir commentaire en dessous pour une suggestion (à réadapter sur l'appel seulement!)

```suggestion
            structureOptionsService.isAverageSkills(idEtablissement)
                        .onSuccess(isAverage -> {
                            ...
                        })
                        .onFailure(err -> {
                            ...
                        })
```